### PR TITLE
autocast for mixed-precision training added to loss_forward

### PR DESCRIPTION
**Pull request by shirinyamani**

### CODE REVIEW

**Comment by shirinyamani** [comment]
This PR adds DDP training, note that DDP expects loss = model(x,y) so loss = model.forward_loss(x,y) is changed! and the step_size is increased to 1000
